### ISSUE TITLE
GSL dependency

### ISSUE THREAD

**Issue by paulgessinger** [AIDASoft/DD4hep#670]
I'm having trouble building DDDigi in a CI build, where I get the following error:

```
CMake Error at DDDigi/CMakeLists.txt:32 (target_include_directories):
  target_include_directories called with incorrect number of arguments
```

Looking at the cmake configuration

https://github.com/AIDASoft/DD4hep/blob/85a0279916039884c8ca0014077c25712617a539/DDDigi/CMakeLists.txt#L17-L32

do I understand that `find_package(GSL)` is never run, the result is then checked, and even if it's **not** found, it tries to use it in the `target_link_libraries` and `target_include_directories` calls?
I believe my error comes from `GSL_INCLUDE_DIR` being empty, which makes the call to `target_include_directories` invalid.

Is this construct on purpose, or could I make the calls for GSL guarded by the if above (which is currently only used for output)?

**Comment by andresailer** [comment]
Which cmake version are you using (just in case it depends on it).
The call to target_include_directories still has a couple of generator expressions in it, so it shouldn't be empty, and thus fulfill the signature for the function.

Can you check if GSL_INCLUDE_DIR has some value?

**Comment by paulgessinger** [comment]
CMake version is 3.17.1. You're right about the generator expressions. Maybe they also evaluate to empty strings? It's a bit difficult for me to get printouts for these values, as I'm trying to build them in a CI run [here](https://github.com/acts-project/ci-dependencies/runs/630513122?check_suite_focus=true) for the ACTS CI Mac build.

**Comment by paulgessinger** [comment]
Ah I just realised I was looking at the master, but I'm building 1.11.2, where the lines mentioned in the error are actually related to TBB, rather than GSL. I'll close and potentially reopen.

**Comment by andresailer** [comment]
The problem with TBB was fixed in #613
